lib/zlib.js

```diff
@@ -432,6 +432,8 @@ function processChunkSync(self, chunk, flushFlag) {
                      availOutBefore); // out_len
     if (error)
       throw error;
+    else if (self[kError])
+      throw self[kError];
```

Comment:
Are we sure this match Streams behavior?

Reply:
No, it does not. Streams do not have "sync" methods. However, zlib creates sync versions of streams that currently `throw`.

Reply:
I think this should throw only for sync versions.

Reply:
Yes, that's what it does. Note this is inside the `processChunkSync` method which is called only for sync versions.